fixing RHCOS URL and clarifying image versions (#15168)
openshift/openshift-docs: modules/installation-user-infra-machines-iso.adoc

.Procedure

. Upload the control plane, compute, and bootstrap Ignition config files that the
installation program created to your HTTP server. Note the URLs of these files.

. Obtain the {op-system} images that are required for your preferred method
of installing operating system instances from the
link:https://access.redhat.com/downloads[Product Downloads] page on the Red
Hat customer portal or the
link:https://mirror.openshift.com/pub/openshift-v4/dependencies/rhcos/4.1/[{op-system} image mirror]
page.
+
[IMPORTANT]
====
The {op-system} images might not change with every release of {product-title}.

> You must download images with the same version number, and that version must be

Reviewer: I think I know what you mean by "You must download images with the same version number" (for the things you're downloading, all should have the same number) however I think that will be confusing in juxtaposition with the next bit. Also they're looking in a versioned directory, and they're only going to download 1 or 2 things so I don't think they'll get confused about the versions of those things matching each other. Best not to say anything?
```suggestion
You must download images with the highest version that is
```

Author: @sosiouxme, @wking, thanks! Is this acceptable?

> less than or equal to the {product-title} version that you install. Use the image versions

Reviewer 2: You should use the largest version that is <= your target release, not any version that is <= your target release.

that match your {product-title} version if they are available.
====
+
You must download the ISO file and either the BIOS or UEFI file.
Those file names resemble the following examples:

** ISO: `rhcos-<version>-installer.iso`
** Compressed metal BIOS: `rhcos-<version>-metal-bios.raw.gz`

> ** Compressed metal UEFI: `rhcos-<version>-metal-uefi.raw.gz`

Reviewer: It's technically `rhcos-<version>-<architecture>-metal-uefi.raw.gz` (e.g. `rhcos-4.1.0-x86_64-metal-uefi.raw.gz`) and similar. This should be obvious to anyone who goes and looks at what's there, so I'm not really concerned about it if you want to roll in version+arch into `<version>`, just wanted to make sure you noticed that.

Author: I fixed it. If we eventually add Power or z support, it'll be an important designation.
